daemon/daemon.go

```diff
@@ -81,6 +81,13 @@ func (d *Daemon) Run() error {
 	writeTicker := time.NewTicker(d.config.WriteInterval)
 	defer writeTicker.Stop()
 
+	labelRefreshInterval := time.Duration(1) * time.Hour
+	if d.config.LabelRefreshInterval > 0 {
+		labelRefreshInterval = d.config.LabelRefreshInterval
+	}
+	labelTicker := time.NewTicker(labelRefreshInterval)
+	defer labelTicker.Stop()
```

Comment:
Are we sure that this is a good change? The original code was IMHO good.

The idea was to stop the timer if `LabelRefreshInterval` is `0`. With this change it will tick every hour as the stop will be run when `Run` method returns.

Reply:
Oh no I literally mentioned the same thing above, then blanked and accepted it anyways. That's on me. I'll revert.
I guess we can still talk about specifc code styles @miyunari .

Reply:
> The idea was to stop the timer if LabelRefreshInterval is 0. With this change it will tick every hour as the stop will be run when Run method returns.

Previously it was for me not clear what it was supposed to do, since we initialized the timer with one hour and stopped it immediately. My proposal was more about the semantics :smiley: 

---

If it is only a matter of stopping the timer in this case, I would only handle the stop itself. Does this suggestion make more sense? :sweat_smile: 
```go
	labelTicker := time.NewTicker(d.config.LabelRefreshIntervaly)
	if d.config.LabelRefreshInterval > 0 {
		defer labelTicker.Stop()
	} else {
		labelTicker.Stop()
	}
```

Reply:
A problem I see with this version is that "d.config.LabelRefreshInterval" could not be set to 0 in this case as time.NewTicker crashes on non-positive values which includes 0.

Thinking about it, it does make sense to throw an error when trying to create a ticker that has a 0 interval.

I guess we could make this work by introducing  a safeguard in config.go that would set all values <= 0 to an arbitrary positive value and check against that in daemon.go as well but I don't know if this is worth it. Do you see a better way of implementing this?

Reply:
Catching it in `config.go` sounds good to me :+1: :smile:

Reply:
I think we are spending too much time on deciding how to save 1 line of code.  :)   

I'm bit afraid that the using the arbitrary value would be less understandable and would lead to even more code.

Reply:
Config.go already parses everything that is provided as a time value into an uint to make sure it is a valid value (so we already have protection in place). 0 is an outlier as it is a valid uint value. 

So if we want to go with your approach  we would need to return a specific value we need to check against, which also needs to be a valid value that is accepted by time.NewTicker which means it needs to be a positive value. 

This is a naive implementation example:
```
        // Config.go
        func parseSeconds(name string, value string, defaultValue time.Duration) (time.Duration, error) {
	        parsedValue, err := strconv.ParseUint(value, 10, 32)
        
	        if err != nil {
		        return defaultValue, fmt.Errorf("invalid value of '%s': %v", name, err.Error())
	        }

                if parsedValue == 0 {
                        return <SOME POSITIVE VALUE>, nil  // <- here we return the new value
                }
        
	        return time.Duration(parsedValue) * time.Second, nil
        }
        // daemon.go
       	labelTicker := time.NewTicker(d.config.LabelRefreshIntervaly) // <- This would finally work
	if d.config.LabelRefreshInterval ==  <SOME POSITIVE VALUE> { // <- here we check against the new value instead of 0
		defer labelTicker.Stop()
	} else {
		labelTicker.Stop()
	}
```
Of course <SOME POSITIVE VALUE> could be a constant that is defined somewhere. 

Now if we want to do something like this we should have this behavior on all of our configurable parameter types to stay consistent.
It also means that this value needs to be chosen in such a way that customers can't select it by accident (or it at least needs to be documented e.g. "valid range 1 - 999999" or something). 

What do you think @pvoborni @miyunari do you want to go down this route?

Reply:
Or in other words: I'll merge this and then we can continue the discussion ;)